### PR TITLE
i#1312 AVX-512 support: Add SYSLOG warning when evex prefix is encountered.

### PR DESCRIPTION
**Pull request by hgreving2304**
Adds a SYSLOG warning if the decoder recognizes an evex encoding until i#1312
will be completed and AVX-512 is fully supported.

Issue: #1312

### CODE REVIEW

File: core/arch/x86/decode.c [hunk 1]
```diff
@@ -741,6 +741,7 @@ read_evex(byte *pc, decode_info_t *di, byte instr_byte,
             return pc;
         }
         *is_evex = true;
+        SYSLOG_INTERNAL_WARNING_ONCE(MSG_AVX_512_SUPPORT_INCOMPLETE_STRING " @" PFX, pc);
```
Comment (derekbruening) [inline core/arch/x86/decode.c]: This is debug-build-only: we want a release-build notice.  SYSLOG takes the name w/o the MSG and no _STRING.  Args are via events.mc specifiers.

Comment (derekbruening) [inline core/arch/x86/decode.c]: If this were 2 months ago, I'd say let's use REPORT_FATAL_ERROR_AND_EXIT.  Should we still do that, with some temporary runtime option to avoid it for your development environment and Travis?  Or just have a printed message?

File: core/win32/events.mc [hunk 2]
```diff
@@ -662,4 +662,12 @@ Language=English
 Application %1!s! (%2!s!). Out of contiguous memory. %3!s!
 .
 
+MessageId =
+Severity = Error
+Facility = DRCore
+SymbolicName = MSG_AVX_512_SUPPORT_INCOMPLETE
+Language=English
+AVX-512 is not yet supported.
```
Comment (derekbruening) [inline core/win32/events.mc]: I would say: "Application %1!s! (%2!s!): AVX-512 was detected at PC %3!s!, but AVX-512 is not yet supported.  Failure is likely."  or sthg?  Maybe reword the 2nd sentence.

File: core/arch/x86/decode.c (hunk 1 above)
Comment (hgreving2304) [reply core/arch/x86/decode.c]: Updated. I looked at the gen_events script. Seemed like it works like this (see update).
In terms of release build, how do I get a release build notice. Doesn't syslog go into the log, and log exists only in debug?

Comment (hgreving2304) [reply core/arch/x86/decode.c]: I'd leave it as a syslog warning so we don't produce an artificial road block. And it's too late now, since it will soon be supported?

File: core/win32/events.mc (hunk 2 above)
Comment (hgreving2304) [reply core/win32/events.mc]: Done.

File: core/arch/x86/decode.c (hunk 1 above)
Comment (derekbruening) [reply core/arch/x86/decode.c]: SYSLOG goes *everywhere*, as specified by {stderr,msgbox,syslog}_mask (and always to log file).  So to show up in release it has to be ERROR not WARNING as you can see from the defaults.

Comment (derekbruening) [reply core/arch/x86/decode.c]: OK.

File: core/win32/events.mc [hunk 3]
```diff
@@ -662,4 +662,12 @@ Language=English
 Application %1!s! (%2!s!). Out of contiguous memory. %3!s!
 .
 
+MessageId =
+Severity = Error
+Facility = DRCore
+SymbolicName = MSG_AVX_512_SUPPORT_INCOMPLETE
+Language=English
+Application %1!s! (%2!s!): AVX-512 was detected at PC 0x%3!lx!
```
Comment (derekbruening) [inline core/win32/events.mc]: First, long is 32 bits in Visual Studio, so this the wrong size specifier.

Second, I don't know for sure whether anything but 's' works for Windows event log -- we had sort of been operating under that assumption as you can see from all of the other numeric params being 's' but I didn't write that code.

File: core/arch/x86/decode.c (hunk 1 above)
Comment (hgreving2304) [reply core/arch/x86/decode.c]: Done.

File: core/win32/events.mc (hunk 3 above)
Comment (hgreving2304) [reply core/win32/events.mc]: That's why I initially didn't use the SYSLOG macro that goes through the gen_events script. Should we remove the @pc message then for now?

Comment (derekbruening) [reply core/win32/events.mc]: Everywhere else we do a local snprintf to a buffer to solve this.

Comment (hgreving2304) [reply core/win32/events.mc]: Sure sure.. Done.

File: core/arch/x86/decode.c [hunk 4]
```diff
@@ -741,6 +741,12 @@ read_evex(byte *pc, decode_info_t *di, byte instr_byte,
             return pc;
         }
         *is_evex = true;
+#if !defined(STANDALONE_DECODER)
+        char pc_addr[IF_X64_ELSE(20, 12)];
+        snprintf(pc_addr, BUFFER_SIZE_ELEMENTS(pc_addr), PFX, pc);
```
Comment (derekbruening) [inline core/arch/x86/decode.c]: style: always NULL_TERMINATE_BUFFER after snprintf

File: core/win32/events.mc [hunk 5]
```diff
@@ -662,4 +662,12 @@ Language=English
 Application %1!s! (%2!s!). Out of contiguous memory. %3!s!
 .
 
+MessageId =
+Severity = Error
+Facility = DRCore
+SymbolicName = MSG_AVX_512_SUPPORT_INCOMPLETE
+Language=English
+Application %1!s! (%2!s!): AVX-512 was detected at PC %3!s!
```
Comment (derekbruening) [inline core/win32/events.mc]: Missing trailing `.`

Comment (derekbruening) [inline core/win32/events.mc]: IMHO this should say something about *why* it matters that AVX-512 was detected.  I think my original suggestion was "expect failure" or sthg: not sure that's great wording, but it should say something: "AVX-512 is not yet fully supported and running applications that use it is not expected to work." or some such.

File: core/arch/x86/decode.c (hunk 4 above)
Comment (hgreving2304) [reply core/arch/x86/decode.c]: Done.

File: core/win32/events.mc (hunk 5 above)
Comment (hgreving2304) [reply core/win32/events.mc]: Done.

Comment (hgreving2304) [reply core/win32/events.mc]: I took that sentence from you above :).
Changed.
